.github/workflows/weekly.yaml

```
@@ -0,0 +1,57 @@
+name: Weekly Tests
+
+on:
+  schedule:
+    - cron: "0 0 * * 6" # Runs every saturday at midnight
+  pull_request:
+    paths:
+      - .github/workflows/weekly.yaml
+permissions:
+  contents: read
+  actions: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cncf:
+    name: CNCF conformance test
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["noble", "jammy"]
+        arch: ["amd64"]
+    runs-on: ['self-hosted', 'linux', '${{ matrix.arch }}', '${{ matrix.os }}', 'xlarge']
+    env:
+      TF_VAR_model: my-canonical-k8s
+    steps:
+    - name: Setup Astral UV
+      uses: astral-sh/setup-uv@v6.7.0
+    - name: Install tox with UV
+      run: uv tool install tox --with tox-uv
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
+      with:
+        provider: lxd
+        channel: latest/stable
+        juju-channel: 3/stable
+    - name: Checking out repo
+      uses: actions/checkout@v5
+    - name: Setup cluster with terraform
+      timeout-minutes: 15
+      run: |
+        tox -e deploy-terraform -- --model ${{env.TF_VAR_model}} --manifest-yaml ${{ github.workspace }}/tests/integration/terraform/default-manifest.yaml
+    - name: Run CNCF conformance test
+      run: |
+        tox -e conformance
```

lets give the conformance test a hint of which juju model to use

```suggestion
        tox -e conformance -- --model ${{env.TF_VAR_model}}
```